test/test_data/http_logging.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "proxy_wasm_intrinsics.h"
+
+class LoggingStreamContext : public Context {
+public:
+  explicit LoggingStreamContext(uint32_t id, RootContext *root) : Context(id, root) {}
+
+  void onCreate() override { LOG_INFO("onCreate called"); }
+  void onDelete() override { LOG_INFO("onDelete called"); }
+  void onDone() override { LOG_INFO("onDone called"); }
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers, bool end_of_stream) override {
+    LOG_INFO("onRequestHeaders called");
+    // A real host would implement header getters/setters to be called here.
```

Comment:
optional: remove comment. The plugin is for testing logging, so I wouldn't expect there to be other hostcalls performed here, and therefore no need to explain their absence.